test/index.test.ts

```diff
@@ -183,15 +236,450 @@ describe('HipThrusTS', () => {
+        // @ts-expect-error
+        const paramAssignableToCorrectShouldFail: ParamAssignableToCorrect = false;
+        // @ts-expect-error
+        const paramAssignableFromCorrectShouldFail: ParamAssignableFromCorrect = false;
+        // @ts-expect-error
+        const returnAssignableToCorrectShouldFail: ReturnAssignableToCorrect = false;
+        // @ts-expect-error
+        const returnAssignableFromCorrectShouldFail: ReturnAssignableFromCorrect = false;
+      });
+      it('attaches properly typed data from right sync data attacher only', () => {
+        const rightPartiallyCovered = {
+          attachData(context: { b: number; other: string }) {
+            return { c: 4 };
+          },
+        };
+
+        const pipedRightOnly = HTPipeAttachData({}, rightPartiallyCovered);
+
+        interface CorrectParam {
+          b: number;
+          other: string;
+        }
+        type CorrectReturnValue = PromiseOrSync<{ c: number }>;
+        type ParamAssignableToCorrect = CorrectParam extends Parameters<
+          typeof pipedRightOnly.attachData
+        >[0]
+          ? true
+          : false;
+        type ParamAssignableFromCorrect = Parameters<
+          typeof pipedRightOnly.attachData
+        >[0] extends CorrectParam
+          ? true
+          : false;
+        type ReturnAssignableToCorrect = CorrectReturnValue extends ReturnType<
+          typeof pipedRightOnly.attachData
+        >
+          ? true
+          : false;
+        type ReturnAssignableFromCorrect = ReturnType<
+          typeof pipedRightOnly.attachData
+        > extends CorrectReturnValue
+          ? true
+          : false;
+        // @ts-expect-error
+        const paramAssignableToCorrectShouldFail: ParamAssignableToCorrect = false;
+        // @ts-expect-error
+        const paramAssignableFromCorrectShouldFail: ParamAssignableFromCorrect = false;
+        // @ts-expect-error
+        const returnAssignableToCorrectShouldFail: ReturnAssignableToCorrect = false;
+        // @ts-expect-error
+        const returnAssignableFromCorrectShouldFail: ReturnAssignableFromCorrect = false;
+      });
+      it('no attaches data when left and right is empty objects', () => {
+        const pipedRightOnly = HTPipeAttachData({}, {});
+
+        // tslint:disable-next-line:prefer-const
+        let returnTypeOfHTPipeAttachData: {};
+
+        // tslint:disable-next-line:prefer-const
+        let typeIsOkay: typeof pipedRightOnly extends typeof returnTypeOfHTPipeAttachData
+          ? true
+          : false = true;
+        // @ts-expect-error
+        typeIsOkay = false;
+      });
+      it('no attaches data when left outputs have type mismatch with right inputs', () => {
+        const leftBad = {
+          attachData(context: { a: string }) {
+            return { b: 'bad' };
+          },
+        };
+        const rightFullyCovered = {
+          attachData(context: { b: number }) {
+            return { c: 4 };
+          },
+        };
+
```

Comment:
- [ ] 1. Where's the async case?
- [ ] 2. Since it's not guaranteed that the HTPipeAttachData won't FAIL at runtime in the javascript world later, wrap this line in a function that never gets called instead.

Reply:
added async case in the end of async tests
wrapped in function that never gets called